### PR TITLE
RAS-1111 Update ROPs to call new survey business end point

### PR DESCRIPTION
**Pull request by LJBabbage**
# What and why?
Updates the call in view_reporting_unit_survey to use the new party service endpoint to find respondents by survey_id and business_id. There is also a small amount of refactoring and removal as well. view_reporting_unit_survey still needs a fair amount of work, so I have left it largely untouched bar the change in this PR

# How to test?
Deploy this PR and seed the db with data. Go to ROPs and search for a Reporting unit, drill down to the survey page of a business and confirm it still works and displays the correct information for respondents
image
Basically the PR is updating the Respondents section of the image above to call from a different endpoint. It is worth adding some more respondents, changing their state, sending a message and doing anything else you can think of
# Jira

### CODE REVIEW

**Review by SteveScorfield — APPROVED** [review]
Changes have been tested based on testing params. Acceptance tests run against PR and all seems good. Code reviewed. Approved.

**Review by arroyoAle — APPROVED** [review]
There is a drop in test coverage which is indirectly caused by this PR as the missed lines are no longer being used by this section and the other section ([view_respondents](https://github.com/ONSdigital/response-operations-ui/blob/085fea1adac7ba57c292f855dc639f9f1334b43a/response_operations_ui/views/reporting_units.py#L109-L125)) has no test coverage, which means we miss lines  223-228, 232-244 of the `party_controller.py`. But not part of this card (maybe worth another card to update the references or add some tests). Otherwise everything looks fine and continues working as expected

**Comment by LJBabbage** [comment]
> There is a drop in test coverage which is indirectly caused by this PR as the missed lines are no longer being used by this section and the other section ([view_respondents](https://github.com/ONSdigital/response-operations-ui/blob/085fea1adac7ba57c292f855dc639f9f1334b43a/response_operations_ui/views/reporting_units.py#L109-L125)) has no test coverage, which means we miss lines 223-228, 232-244 of the `party_controller.py`. But not part of this card (maybe worth another card to update the references or add some tests). Otherwise everything looks fine and continues working as expected

Difficult one, might not be used anymore, but as it's an endpoint hard to track down
